guides/migrations/2.0.x_2.1.0.md

```diff
@@ -0,0 +1,125 @@
+# Migration guide from 2.0.x to 2.1.0
+There are a few changes necessary in the configuration files and a few additional, optional ones.
+## Main config (`config/config.exs`)
+MongoosePush supports structured logging in 2.1.0 release. To use that feature, there is a need to change logger configuration to use the new default.
+Old logger config:
+
+```elixir
+config :logger, :console,
+  format: "\n$dateT$time [$level] $metadata$levelpad$message\n",
+  metadata: [:pid]
+```
+
+New logger config:
+```elixir
+config :logger, :console,
+  format: {MongoosePush.Logger.LogFmt, :format},
+  metadata: :all
+```
+Further notes on logging are in the [logging section](2.0.x_2.1.0.md#logging)
+
+Additionally there are a few new defaults needed to be set up:
+```elixir
+config :mongoose_push, MongoosePush.Service,
+  fcm: MongoosePush.Service.FCM,
+  apns: MongoosePush.Service.APNS
+
+config :mongoose_push, backend_module: MongoosePush
+
+config :phoenix, :json_library, Jason
+```
+
+Lastly, since 2.1.0 MongoosePush uses `Phoenix` instead of `Maru` and Maru config entry can be removed.
+```elixir
+config :maru, :test, false
+```
+
+## Environmental configs (`config/{prod|dev|test}.exs`)
+
+### `Phoenix` in place of `Maru`
+As mentioned earlier, MongoosePush uses `Phoenix` and the config has to be updated to reflect that change.
+Old `Maru` config entry:
+```elixir
+config :maru, MongoosePush.Router,
+  versioning: [
+    using: :path
+  ],
+  https: [
+    bind_addr: {:system, :string, "PUSH_HTTPS_BIND_ADDR", "127.0.0.1"},
+    port: {:system, :integer, "PUSH_HTTPS_PORT", 8443},
+    keyfile: {:system, :string, "PUSH_HTTPS_KEYFILE", "priv/ssl/fake_key.pem"},
+    certfile: {:system, :string, "PUSH_HTTPS_CERTFILE", "priv/ssl/fake_cert.pem"},
+    cacertfile: {:system, :string, "PUSH_HTTPS_CERTFILE", "priv/ssl/fake_cert.pem"},
+    acceptors: {:system, :integer, "PUSH_HTTPS_ACCEPTORS", 100},
+    otp_app: :mongoose_push
+  ]
+```
+It should be removed completely and in its place there must be the new, `Phoenix` config entry:
+```elixir
+config :mongoose_push, MongoosePushWeb.Endpoint,
+  https: [
+    ip: {
+      :system,
+      # Custom type parser (Phoenix needs erlang-inet-style IP address)
+      {MongoosePush.Config.Utils, :parse_bind_addr, []},
+      "PUSH_HTTPS_BIND_ADDR",
+      {127, 0, 0, 1}
+    },
+    port: {:system, :integer, "PUSH_HTTPS_PORT", 8443},
+    keyfile: {:system, :string, "PUSH_HTTPS_KEYFILE", "priv/ssl/fake_key.pem"},
+    certfile: {:system, :string, "PUSH_HTTPS_CERTFILE", "priv/ssl/fake_cert.pem"},
+    cacertfile: {:system, :string, "PUSH_HTTPS_CERTFILE", "priv/ssl/fake_cert.pem"},
+    protocol_options: [
+      # https://ninenines.eu/docs/en/cowboy/2.5/manual/cowboy_http/
```

Comment:
I am almost ready to bet that on cowboy update somebody will forget to update this link ;)